### PR TITLE
Layer handling simplified, and streamlined

### LINKED ISSUE #15: Handle layer switcher keys the right way
There are 4 layers per keymap. The base layer is active by default when no layer switcher key is pressed. There are also the Mod, Mouse, and Fn layers activated by the Mod, Mouse, and Fn keys respectively.

When multiple layer switcher keys are pressed the first one dominates. I mean the one that gets pressed before the others. This allows us to switch to the Mouse layer by holding the Mouse key pressed, then also hold the Mod key to accelerate the movement of the pointer according to [the factory keymap](https://ultimatehackingkeyboard.com/#layers).

### PR DESCRIPTION
**Pull request by algernon**
This simplifies the layer handling by only allowing a single layer to be active at any one time. As such, the behaviour after this patch should be as described, and as discussed in #15, as far as I understand.

Also moved the layer handling to a separate file, let me know if that is not desired.

Fixes #15, if all goes well.

### CODE REVIEW

File: right/src/layer.c [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#include "layer.h"
+
+uint8_t ActiveLayer = LAYER_ID_BASE;
+
+void Layer_MoveTo(uint8_t layer) {
```
Comment (mondalaci) [inline right/src/layer.c]: I would rather leave out these helper functions. They don't seem to add to the readability but make our internal API more complex.

**Review by mondalaci — CHANGES_REQUESTED** [review]

File: right/src/layer.c (hunk 1 above)
Comment (algernon) [reply right/src/layer.c]: A'ight. Update coming up in a bit then.

**Comment by algernon** [comment]
Updated, should be much simpler now.

**Comment by mondalaci** [comment]
Looks great, thanks!

**Pull request merged**